crates/router/src/connector/datatrans/transformers.rs

```diff
@@ -1,235 +1,519 @@
-use masking::Secret;
+use std::fmt::Debug;
+
+use base64::Engine;
+use common_utils::types::MinorUnit;
+use masking::{PeekInterface, Secret};
 use serde::{Deserialize, Serialize};
 
 use crate::{
-    connector::utils::PaymentsAuthorizeRequestData,
+    connector::utils::{PaymentsAuthorizeRequestData, RouterData},
+    consts,
     core::errors,
-    types::{self, api, domain, storage::enums},
+    types::{
+        self,
+        api::{self, enums},
+        domain,
+        transformers::ForeignFrom,
+    },
 };
 
-//TODO: Fill the struct with respective fields
+#[derive(Default, Debug, Clone, Serialize, Deserialize)]
+pub struct DatatransErrorResponse {
+    pub error: DatatransError,
+}
+pub struct DatatransAuthType {
+    pub(super) auth_header: String,
+}
+
 pub struct DatatransRouterData<T> {
-    pub amount: i64, // The type of amount that a connector accepts, for example, String, i64, f64, etc.
+    pub amount: MinorUnit,
     pub router_data: T,
 }
 
-impl<T> TryFrom<(&api::CurrencyUnit, enums::Currency, i64, T)> for DatatransRouterData<T> {
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct DatatransPaymentsRequest {
+    pub amount: MinorUnit,
+    pub currency: enums::Currency,
+    pub card: PlainCardDetails,
+    pub refno: String,
+    pub auto_settle: bool,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "snake_case")]
+pub enum TransactionType {
+    Payment,
+    Credit,
+    CardCheck,
+}
+#[derive(Debug, Deserialize, Serialize)]
```

Comment:
If we are not sending this in the request, no need to derive `Serialize`. Please resolve this at all the possible places. Same goes for `Deserialize`